tests/unit/dataloader/test_torch_dataloader.py

```diff
@@ -366,3 +378,43 @@ def add_sample_weight(features, labels, sample_weight_col_name="sample_weight"):
         assert list(X["cont2"].cpu().numpy()) == [2.0] * 10
 
         assert list(sample_weight.cpu().numpy()) == [1.0] * 10
+
+
+@pytest.mark.parametrize("lists_as_tuple", [True, False])
+@pytest.mark.parametrize("tensors_as_1d", [True, False])
+def test_1d_tensors(tensors_as_1d, lists_as_tuple):
+    # create small dataset, add values to sparse_list
+    df = make_df(
+        {
+            "cat1": [1] * 4,
+            "cat2": [2] * 4,
+            "cat3": [3] * 4,
+            "cont2": [2.0] * 4,
+            "spar1": [[1, 2, 3, 4], [4, 2, 4, 4], [1, 3, 4, 3], [1, 1, 3, 3]],
+            "spar2": [[1, 2, 3, 4, 5], [6, 7, 8, 9, 10], [11, 12, 13, 14], [15, 16]],
+        }
+    )
+    spa_lst = ["spar1", "spar2"]
+    batch_size = 2
+
+    ds = Dataset(df)
+    dataloader = torch_dataloader.Loader(
+        ds,
+        batch_size=batch_size,
+        lists_as_tuple=lists_as_tuple,
+        tensors_as_1d=tensors_as_1d
+    )
+    for batch in dataloader:
+        feats, labs = batch
+        for col in feats.keys():
+            if lists_as_tuple and col in spa_lst:
+                feature_tensor = feats[col][0]
+            else:
+                feature_tensor = feats[col]
+            print(col)
+            print(feature_tensor)
+            if tensors_as_1d or "__values" in col or "__offsets" in col or col in spa_lst:
+                assert len(list(feature_tensor.shape)) == 1
```

Comment:
Might be good to add test assertions about the offsets either in this test or in a new test, just to double-check that the resulting offsets are what we expect